aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/SmokeTestsDecorator.kt

```diff
@@ -0,0 +1,172 @@
+                            val instantiator = SmokeTestsInstantiator(codegenContext)
+                            instantiator.renderConf(this, testCase.vendorParams)
+                            rust("let client = crate::Client::from_conf(conf);")
+                            instantiator.renderInput(this, operation, operationInput, testCase.params)
+                            instantiator.renderExpectation(this, testCase.expectation)
+                        }
+                    }
+                }
+
+            else -> emptySection
+        }
+    }
+}
+
+class SmokeTestsBuilderKindBehavior(val codegenContext: CodegenContext) : Instantiator.BuilderKindBehavior {
+    override fun hasFallibleBuilder(shape: StructureShape): Boolean =
+        BuilderGenerator.hasFallibleBuilder(shape, codegenContext.symbolProvider)
+
+    override fun setterName(memberShape: MemberShape): String = memberShape.setterName()
+
+    override fun doesSetterTakeInOption(memberShape: MemberShape): Boolean = true
+}
+
+class SmokeTestsInstantiator(private val codegenContext: ClientCodegenContext) : Instantiator(
+    codegenContext.symbolProvider,
+    codegenContext.model,
+    codegenContext.runtimeConfig,
+    SmokeTestsBuilderKindBehavior(codegenContext),
+) {
+    fun renderConf(
+        writer: RustWriter,
+        data: Optional<ObjectNode>,
+    ) {
+        writer.rust("let conf = crate::config::Builder::new()")
+        writer.indent()
+        writer.rust(".behavior_version(crate::config::BehaviorVersion::latest())")
+        data.orNull()?.let { node ->
+            val region = node.getStringMemberOrDefault("region", "us-west-2")
+//            val sigv4aRegionSet = node.getArrayMember("sigv4aRegionSet")
+//                .map { a ->
+//                    a.getElementsAs { el ->
+//                        el.expectStringNode().getValue()
+//                    }
+//                }
+//                .orElse(null)
+//            val useAccountIdRouting = node.getBooleanMemberOrDefault("useAccountIdRouting", true)
+            val useDualstack = node.getBooleanMemberOrDefault("useDualstack", false)
+            val useFips = node.getBooleanMemberOrDefault("useFips", false)
+            val uri = node.getStringMemberOrDefault("uri", null)
+            val useAccelerate = node.getBooleanMemberOrDefault("useAccelerate", false)
+//            val useGlobalEndpoint = node.getBooleanMemberOrDefault("useGlobalEndpoint", false)
+            val forcePathStyle = node.getBooleanMemberOrDefault("forcePathStyle", false)
+            val useArnRegion = node.getBooleanMemberOrDefault("useArnRegion", true)
+            val useMultiRegionAccessPoints = node.getBooleanMemberOrDefault("useMultiRegionAccessPoints", true)
+
+            region?.let { writer.rust(".region(crate::config::Region::new(${it.dq()}))") }
+//            sigv4aRegionSet?.let { writer.rust("._($it)") }
+//            useAccountIdRouting?.let { writer.rust("._($it)") }
```

Comment:
We don't support this one yet

Reply:
I was thinking maybe we should add `TODO` for commented lines, but am curious whether you can think of a better way so that when account id based routing is supported for instance, it helps us remember to revisit here and comment in corresponding lines?

Reply:
I added a "throw error" for these so we'll know when we start getting test cases that require setting these.